src/daft-functions/src/binary/slice.rs

```diff
@@ -0,0 +1,66 @@
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    prelude::{DataType, Field, Schema},
+    series::Series,
+};
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct BinarySlice {}
+
+#[typetag::serde]
+impl ScalarUDF for BinarySlice {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+    fn name(&self) -> &'static str {
+        "binary_slice"
+    }
+
+    fn to_field(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
+        match inputs {
+            [data, start, length] => {
+                let data = data.to_field(schema)?;
+                let start = start.to_field(schema)?;
+                let length = length.to_field(schema)?;
+
+                match &data.dtype {
+                    DataType::Binary | DataType::FixedSizeBinary(_) => {
+                        if start.dtype.is_integer() && (length.dtype.is_integer() || length.dtype.is_null()) {
+                            Ok(Field::new(data.name, DataType::Binary))
+                        } else {
+                            Err(DaftError::TypeError(format!(
+                                "Expects inputs to binary_slice to be binary, integer and integer or null but received {}, {} and {}",
+                                data.dtype, start.dtype, length.dtype
+                            )))
+                        }
+                    }
+                    _ => Err(DaftError::TypeError(format!(
+                        "Expects inputs to binary_slice to be binary, integer and integer or null but received {}, {} and {}",
+                        data.dtype, start.dtype, length.dtype
+                    ))),
+                }
+            }
+            _ => Err(DaftError::SchemaMismatch(format!(
+                "Expected 3 input args, got {}",
+                inputs.len()
+            ))),
+        }
+    }
+
+    fn evaluate(&self, inputs: &[Series]) -> DaftResult<Series> {
```

Comment:
same comment as concat. also you could match on the length of inputs and ensure that proper error handling occurs if the length does not match what is expected